providers/asrockrack/asrockrack.go

```diff
@@ -0,0 +1,304 @@
+package asrockrack
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/bmc-toolbox/bmclib/internal/httpclient"
+	"github.com/bmc-toolbox/bmclib/providers"
+	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/registrar"
+)
+
+const (
+	// ProviderName for the provider implementation
+	ProviderName = "asrockrack"
+	// ProviderProtocol for the provider implementation
+	ProviderProtocol = "vendorapi"
+)
+
+var (
+	// Features implemented by asrockrack https
+	Features = registrar.Features{
+		providers.FeatureBiosVersionRead,
+		providers.FeatureBmcVersionRead,
+		providers.FeatureBiosFirmwareUpdate,
+		providers.FeatureBmcFirmwareUpdate,
+	}
+)
+
+// ASRockRack holds the status and properties of a connection to a asrockrack bmc
+type ASRockRack struct {
+	ip            string
+	username      string
+	password      string
+	loginSession  *loginSession
+	httpClient    *http.Client
+	fwInfo        *firmwareInfo
+	resetRequired bool // Indicates if the BMC requires a reset
+	skipLogout    bool // A Close() / httpsLogout() request is ignored if the BMC was just flashed - since the sessions are terminated either way
+	log           logr.Logger
+}
+
+// New returns a new ASRockRack instance ready to be used
+func New(ctx context.Context, ip string, username string, password string, log logr.Logger) (*ASRockRack, error) {
```

Comment:
`ctx` doesn't look used. could be removed.

Reply:
fixed in 3a22d5e102